protocol/chainlib/chainproxy/connector_test.go

```diff
@@ -2,26 +2,29 @@ package chainproxy
 
 import (
 	"context"
+	"fmt"
 	"log"
 	"net"
 	"net/http"
 	"net/rpc"
+	"os"
 	"testing"
 	"time"
 
 	"github.com/lavanet/lava/protocol/chainlib/chainproxy/rpcclient"
 	"github.com/lavanet/lava/protocol/common"
+	"github.com/lavanet/lava/utils"
 	pb_pkg "github.com/lavanet/lava/x/spec/types"
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 )
 
 const (
-	listenerAddress    = "localhost:1234"
-	port               = "1234"
-	listenerAddressTcp = "http://localhost:1234"
-	numberOfClients    = 5
+	listenerAddress     = "localhost:1234"
+	listenerAddressGrpc = "localhost:1235"
```

Comment:
let's use a generic port :0 like we do in our more advanced tests, it's easy to do a listen to port 0 then get the port and save it in a global variable, continue form there